### PR TITLE
Improve support for crash reports with Authenticated Pointers on arm64e

### PR DESCRIPTION
**Pull request by lumaxis**

### CODE REVIEW

File: Classes/BITCrashReportTextFormatter.m [hunk 1]
```diff
@@ -830,11 +833,16 @@ + (NSString *)bit_formatStackFrame: (BITPLCrashReportStackFrameInfo *) frameInfo
   NSString *imageName = @"\?\?\?";
   NSString *symbolString = nil;
   
-  BITPLCrashReportBinaryImageInfo *imageInfo = [report imageForAddress: frameInfo.instructionPointer];
+  // When on an ARM64 architecture, normalize the address to remove possible pointer signatures
+  uint64_t instructionPointer = lp64 ? (frameInfo.instructionPointer & 0x0000000fffffffff) : frameInfo.instructionPointer;
```
Comment (ElektrojungeMS) [inline Classes/BITCrashReportTextFormatter.m]: can we rename this to `normalizedInstructionPointer`, please?

Comment (lumaxis) [reply Classes/BITCrashReportTextFormatter.m]: Hah, I even had that at some point, I believe 🙂 Will update!

File: Classes/BITCrashReportTextFormatter.m [hunk 2]
```diff
@@ -830,11 +833,16 @@ + (NSString *)bit_formatStackFrame: (BITPLCrashReportStackFrameInfo *) frameInfo
...
+  uint64_t untouchedInstructionPointer = frameInfo.instructionPointer;
+  
+  BITPLCrashReportBinaryImageInfo *imageInfo = [report imageForAddress: instructionPointer];
```
Comment (lumaxis) [inline Classes/BITCrashReportTextFormatter.m]: We might be able to ultimately improve this and move the above logic into PLCrashReporter.

**Comment by lumaxis** [comment]
@ElektrojungeMS Updated!

**Pull request merged**